enarx-keep-sgx/src/layout.rs

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::component::Component;
+
+use span::{Contains as _, Line, Span};
+
+const PREFIX: usize = units::bytes![4; MiB];
+const ALIGN: usize = units::bytes![2; MiB];
+const STACK: usize = units::bytes![8; MiB];
+const HEAP: usize = units::bytes![128; MiB];
+const SIZE: usize = units::bytes![64; GiB];
+
+const fn lower(value: usize, boundary: usize) -> usize {
+    value / boundary * boundary
+}
+
+const fn raise(value: usize, boundary: usize) -> usize {
+    lower(value + boundary - 1, boundary)
+}
+
+fn above(rel: impl Into<Line<usize>>, size: usize) -> Span<usize> {
+    Span {
+        start: raise(rel.into().end, ALIGN),
+        count: size,
+    }
+}
+
+fn below(rel: impl Into<Line<usize>>, size: usize) -> Span<usize> {
+    Span {
+        start: lower(rel.into().start - size, ALIGN),
+        count: size,
+    }
+}
+
+// NOTE: this structure MUST be kept in sync with enarx-keep-sgx-shim
```

Comment:
Should this be a debt issue? I can file it.

Reply:
Yes, please do.

Reply:
https://github.com/enarx/enarx/issues/463